### PR TITLE
SC-12905 improve and add metric descriptions

### PR DESCRIPTION
**Pull request by yagogou**

### CODE REVIEW

**Review by cpipilas — APPROVED** [review]
LGTM

File: docs/integration/kubernetes.md [hunk 1]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
-|kubelet.http_request.p75latency|float64|P75 HTTP request latencies in microseconds|
-|kubelet.http_request.p90latency|float64|P90 HTTP request latencies in microseconds|
-|kubelet.http_request.p95latency|float64|P95 HTTP request latencies in microseconds|
-|kubelet.http_request.p99latency|float64|P99 HTTP request latencies in microseconds|
+|Metric Name|Unit| Description                                |
+|----|----|--------------------------------------------|
+|kubelet.http_request.operations|float64| total number of HTTP request operations    |
+|kubelet.http_request.total_num|int64| total number of HTTP requests              |
+|kubelet.http_request.duration|float64| HTTP request duration                      |
+|kubelet.http_request.p50latency|float64| p50 HTTP request latencies in seconds      |
```
Comment (otisg) [inline docs/integration/kubernetes.md:330]: latencies ==> latency here and below?

File: docs/integration/kubernetes.md [hunk 2]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+|----|----|------------------------------------------------------------|
+|apiserver.inflight_requests.total_num|uint64| number of in-flight requests being processed by API server |
 
 #### Audit Events
-|Metric Name|Unit|Description|
-|----|----|-----------|
-|apiserver.audit_events.total_num|uint64|Counter of audit events generated and sent to the audit backend|
+|Metric Name|Unit| Description                                                     |
+|----|----|-----------------------------------------------------------------|
+|apiserver.audit_events.total_num|uint64| counter of audit events generated and sent to the audit backend |
```
Comment (otisg) [inline docs/integration/kubernetes.md:388]: counter ==> count or number?

File: docs/integration/kubernetes.md [hunk 3]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+|apiserver.key_gen_duration.duration|float64| duration of key generation for storing data in the API server's storage in seconds |
+|apiserver.key_gen_duration.total_num|uint64| number of key generation for storing data in the API server's storage              |
 
 #### Requests Total
-|Metric Name|Unit|Description|
-|----|----|-----------|
-|apiserver.request_total.total_num|uint64|Counter of apiserver requests|
+|Metric Name|Unit| Description                  |
+|----|----|------------------------------|
+|apiserver.request_total.total_num|uint64| number of apiserver requests |
```
Comment (otisg) [inline docs/integration/kubernetes.md:488]: apiserver ==> API server (we use that below)

File: docs/integration/kubernetes.md [hunk 4]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
-|etcd.disk_backend_commit.total_num|uint64||
-|etcd.disk_backend_commit.duration|float64|The latency distributions of commit called by backend|
+|Metric Name|Unit| Description                       |
+|----|----|-----------------------------------|
+|etcd.disk_backend_commit.p50latency|float64| p50 latency of commit operations  |
+|etcd.disk_backend_commit.p75latency|float64| p75 latency of commit operations  |
+|etcd.disk_backend_commit.p90latency|float64| p90 latency of commit operations  |
+|etcd.disk_backend_commit.p95latency|float64| p95 latency of commit operations  |
+|etcd.disk_backend_commit.p99latency|float64| p99 latency of commit operations  |
+|etcd.disk_backend_commit.total_num|uint64| number of commit operations for etcd's disk bakend commits |
```
Comment (otisg) [inline docs/integration/kubernetes.md:521]: bakend ==> backend

File: docs/integration/kubernetes.md [hunk 5]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
-|etcd.disk_wal_fsync.total_num|uint64||
-|etcd.disk_wal_fsync.duration|float64|The latency distributions of fsync called by WAL|
+|Metric Name|Unit| Description                      |
+|----|----|----------------------------------|
+|etcd.disk_wal_fsync.p50latency|float64| p50 latency of fsync operations  |
+|etcd.disk_wal_fsync.p75latency|float64| p75 latency of fsync operations  |
+|etcd.disk_wal_fsync.p90latency|float64| p90 latency of fsync operations  |
+|etcd.disk_wal_fsync.p95latency|float64| p95 latency of fsync operations  |
+|etcd.disk_wal_fsync.p99latency|float64| p99 latency of fsync operations  |
+|etcd.disk_wal_fsync.total_num|uint64| number of fsync (flush to disk) operations for etcd's write-ahead log (WAL) on the disk backend |
```
Comment (otisg) [inline docs/integration/kubernetes.md:532]: Does "disk backend" make sense?  Why not just "disk"?

File: docs/integration/kubernetes.md [hunk 6]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
-|etcd.disk_backend_snapshot.total_num|uint64||
-|etcd.disk_backend_snapshot.duration|float64|The latency distribution of backend snapshots|
+|Metric Name|Unit| Description                                                      |
+|----|----|------------------------------------------------------------------|
+|etcd.disk_backend_snapshot.p50latency|float64| p50 latency of snapshot creation                                 |
+|etcd.disk_backend_snapshot.p75latency|float64| p75 latency of snapshot creation                                 |
+|etcd.disk_backend_snapshot.p90latency|float64| p90 latency of snapshot creation                                 |
+|etcd.disk_backend_snapshot.p95latency|float64| p95 latency of snapshot creation                                 |
+|etcd.disk_backend_snapshot.p99latency|float64| p99 latency of snapshot creation                                 |
+|etcd.disk_backend_snapshot.total_num|uint64| number of snapshot creation for etcd's disk backend              |
```
Comment (otisg) [inline docs/integration/kubernetes.md:543]: disk backend...

File: docs/integration/kubernetes.md [hunk 7]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
-|etcd.store.watchers|uint64|Count of currently active watchers|
+|Metric Name| Unit  | Description                                                                                 |
+|----|-------|---------------------------------------------------------------------------------------------|
+|etcd.mvcc.keys| uint64| number of keys                                                                              |
+|etcd.mvcc.events| uint64| number of events sent by this member                                                        |
+|etcd.mvcc.keys_bytes| uint64| number of MVCC keys involved in etcd debugging operations                                   |
+|etcd.mvcc.watch_stream| uint64| number of watch streams                                                                     |
+|etcd.mvcc.watcher| uint64| number of watchers                                                                          |
+|etcd.mvcc.db_read| uint64| number of currently open read transactions                                                  |
+|etcd.mvcc.db_size| uint64| size of the underlying database physically allocated in bytes.                              |
```
Comment (otisg) [inline docs/integration/kubernetes.md:560]: trailing period not needed

File: docs/integration/kubernetes.md [hunk 8]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
 
 ### Storage Metrics
 
-|Name|Type|Unit|Numeric Type|Label|Description|
-|----|----|----|------------|-----|-----------|
-|kubernetes.pvc.available|gauge|bytes|long|available bytes|number of available bytes in the volume|
-|kubernetes.pvc.used|gauge|bytes|long|used bytes|number of used bytes in the volume|
-|kubernetes.pvc.capacity|gauge|bytes|long|volume capacity|the capacity in bytes of the volume|
+|Name|Type|Unit|Numeric Type|Label| Description                           |
```
Comment (otisg) [inline docs/integration/kubernetes.md:195]: This table has more columns

File: docs/integration/kubernetes.md [hunk 9]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+|kubernetes.pvc.capacity|gauge|bytes|long|volume capacity|the capacity in bytes of the volume    |
 
 ### Kubelet Metrics
 #### Volume Stats
-|Metric Name|Unit|Description|
-|----|----|-----------|
-|pvc.used|int64|Number of used inodes in the volume|
-|pvc.available|int64|Number of available bytes in the volume|
-|pvc.capacity|int64|Capacity in bytes of the volume|
+|Metric Name|Unit| Description                           |
```
Comment (otisg) [inline docs/integration/kubernetes.md:203]: int64, etc. etc. (see all other tables below) are not "Unit".  "Unit" was meant like "seconds", "bytes", "count" and such.
.... then we don't have to include things like ".... in seconds" in the description....

File: docs/integration/kubernetes.md [hunk 10]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
-|etcd.mvcc_db_compaction.duration|float64||
+|Metric Name|Unit| Description                                                      |
+|----|----|------------------------------------------------------------------|
+|etcd.mvcc_db_compaction.p50latency|float64| p50 latency of MVCC database compactation operations             |
+|etcd.mvcc_db_compaction.p75latency|float64| p75 latency of MVCC database compactation operations             |
+|etcd.mvcc_db_compaction.p90latency|float64| p90 latency of MVCC database compactation operations             |
+|etcd.mvcc_db_compaction.p95latency|float64| p95 latency of MVCC database compactation operations             |
+|etcd.mvcc_db_compaction.p99latency|float64| p99 latency of MVCC database compactation operations             |
+|etcd.mvcc_db_compaction.total_num|uint64| number of MVCC database compactation operations                  |
+|etcd.mvcc_db_compaction.duration|float64| duration of MVCC database compactation operations in miliseconds |
```
Comment (otisg) [inline docs/integration/kubernetes.md:599]: compactation ==> compaction.  See above, too.

File: docs/integration/kubernetes.md [hunk 11]
```diff
@@ -192,598 +192,593 @@ Container and Kubernetes metrics are collected along with labels and tags, which
...
+|coredns.forward_request.total_num|uint64| number of forwarding DNS requests handled by CoreDNS             |
 
 ### Scheduler Metrics
 #### Preemption Attempts
-|Metric Name|Unit|Description|
-|----|----|-----------|
-|scheduler.preemption_attempts.total_num|uint64|Total preemption attempts in the cluster till now|
+|Metric Name|Unit| Description                                       |
+|----|----|---------------------------------------------------|
+|scheduler.preemption_attempts.total_num|uint64| number preemption attempts in the cluster         |
```
Comment (otisg) [inline docs/integration/kubernetes.md:696]: number ==> number of

**Review by otisg — CHANGES_REQUESTED** [review]
Good work.  Left some comments.

File: docs/integration/kubernetes.md (hunk 5 above)
Comment (yagogou) [reply docs/integration/kubernetes.md:532]: The metric is named that way `disk backend`, for example we also have `disk_wal` in this case, if we name just disk can be confusing IMO

File: docs/integration/kubernetes.md (hunk 6 above)
Comment (yagogou) [reply docs/integration/kubernetes.md:543]: I can change if you disagree with my response from https://github.com/sematext/docs/pull/514#discussion_r1293183122

File: docs/integration/kubernetes.md (hunk 8 above)
Comment (yagogou) [reply docs/integration/kubernetes.md:195]: I don't really see that the `Label` column is useful at all, removed the column

File: docs/integration/kubernetes.md (hunk 9 above)
Comment (yagogou) [reply docs/integration/kubernetes.md:203]: so, I change all the tables with the real unit?

**Comment by yagogou** [comment]
@otisg see the comments that I left

File: docs/integration/kubernetes.md (hunk 9 above)
Comment (otisg) [reply docs/integration/kubernetes.md]: @cpipilas ? (out for the rest of the week)

Comment (cpipilas) [reply docs/integration/kubernetes.md]: We can add `Type` and `Unit`.  
`Type` can include things like gauge, long gauge, counter, etc. 
`Unit` can include bytes, milliseconds, etc.

Comment (yago-123) [reply docs/integration/kubernetes.md]: created ticket https://sematext.atlassian.net/browse/SC-17219

**Review by cpipilas — APPROVED** [review]
LGTM given that we will update unit as planned.

**Comment by yago-123** [comment]
Leaving the PR open until the review from @otisg, just review the last two commits regarding runtime metrics

File: docs/integration/kubernetes.md [hunk 12]
```diff
@@ -780,6 +780,24 @@ Container and Kubernetes metrics are collected along with labels and tags, which
 |----|----|-------------------------------------|
 |scheduler.attempts.total_num|uint64| number of attempts to schedule pods |
 
+### Runtime Metrics
+| Metric Name                            |Unit| Description                       |
+|----------------------------------------|----|-----------------------------------|
+| kubernetes.runtime.os_threads          |uint64| number of operating system threads used |
+| kubernetes.runtime.goroutines          |uint64| number of goroutines active  |
+| kubernetes.runtime.resident_memory     |uint64| amount of memory for process's code and data occupied |
```
Comment (otisg) [inline docs/integration/kubernetes.md]: process's ==> process'

**Review by otisg — COMMENTED** [review]
Just 1 minime thing

File: docs/integration/kubernetes.md (hunk 12 above)
Comment (yago-123) [reply docs/integration/kubernetes.md]: commited to master, the commit does not show up in this PR by some reason

**Pull request merged**